src/models/highlightedtext.h

```diff
@@ -0,0 +1,67 @@
+/*
+    SPDX-FileCopyrightText: Lieven Hey <lieven.hey@kdab.com>
+    SPDX-FileCopyrightText: 2023 Klarälvdalens Datakonsult AB, a KDAB Group company, info@kdab.com
+
+    SPDX-License-Identifier: GPL-2.0-or-later
+*/
+
+#pragma once
+
+#include <QObject>
+#include <QTextLayout>
+#include <QTextLine>
+
+#include <KColorScheme>
+
+#include "hotspot-config.h"
+
+namespace KSyntaxHighlighting {
+class SyntaxHighlighter;
+class Definition;
+class Repository;
+}
+
+class HighlightingImplementation
+{
+    Q_DISABLE_COPY(HighlightingImplementation)
+public:
+    HighlightingImplementation() = default;
+    virtual ~HighlightingImplementation() = default;
+
+    virtual QVector<QTextLayout::FormatRange> format(const QStringList& text) = 0;
```

Comment:
why is this a pure virtual interface? there is only one implementation at compile time - remove all the virtuals here from the header

Reply:
also, this class is not public - meaning people should only deal with HighlightedText below which uses it internally but otherwise doesn't provide a getter to this class. As such, forward declare this here in the header and move all code to the .cpp